libdnf/rpm/package_query.cpp

```diff
@@ -2148,6 +2148,30 @@ static void add_n_first_to_map(
     }
 }
 
+static void add_duplicates_to_map(
+    libdnf::solv::Pool & pool,
+    libdnf::solv::SolvMap & result,
+    libdnf::solv::IdQueue & samename,
+    int start_block,
+    int stop_block) {
+    Solvable * s_first = nullptr;
+    Solvable * s_second = nullptr;
+    for (int pos_first = start_block; pos_first < stop_block; ++pos_first) {
+        Id id_first = samename[pos_first];
+        s_first = pool.id2solvable(id_first);
+        for (int pos_second = start_block; pos_second < stop_block; ++pos_second) {
```

Comment:
You compare each couple of solvables twice (i.e. pkgA with pkgB and than pkgB with pkgA again). I think you can start the inner cycle with `for (int pos_second = pos_first...`. Or even `pos_second = pos_first + 1` to avoid comparison of solvable with itself.

Reply:
Yes, thank you!